skynet.go

```diff
@@ -8,62 +8,85 @@ import (
 	"mime/multipart"
 	"net/http"
 	"os"
+	gopath "path"
 	"path/filepath"
 	"strings"
+
+	"gitlab.com/NebulousLabs/errors"
 )
 
 type (
-	UploadReponse struct {
-		Skylink string `json:"skylink"`
-	}
-
+	// UploadOptions contains the options used for uploads.
 	UploadOptions struct {
-		portalUrl                    string
-		portalUploadPath             string
-		portalFileFieldname          string
-		portalDirectoryFileFieldname string
-		customFilename               string
+		// PortalURL is the URL of the portal to use.
+		PortalURL string
+		// PortalUploadPath is the path on the portal of the upload endpoint.
+		PortalUploadPath string
+		// PortalFileFieldname is the fieldname for files on the portal.
+		PortalFileFieldname string
+		// PortalDirectoryFileFieldname is the fieldname for directory files on
+		// the portal.
+		PortalDirectoryFileFieldname string
+		// CustomFilename is the custom filename to use for the upload. If this
+		// is empty, the filename of the file being uploaded will be used by
+		// default.
+		CustomFilename string
```

Comment:
I think we should capitalize `name` in these. `fieldname` is not really a word, so the correct camel style name should be `fieldName`.